uniffi_bindgen/src/scaffolding/templates/macros.rs

```diff
@@ -6,6 +6,35 @@
 r#{{ func.name() }}({% call _arg_list_rs_call(func) -%})
 {%- endmacro -%}
 
+{%- macro to_rs_method_call(obj_name, meth) -%}
+{#  Some magic methods need a uniffi-compatible shim #}
+{% if meth.magic() == "hash" %}
+    {
+        // the magic method "hash" - described as `fn(&self) -> i64`
+        use ::std::hash::{Hash, Hasher};
+        let mut s = ::std::collections::hash_map::DefaultHasher::new();
+        match<std::sync::Arc<{{ obj_name }}> as ::uniffi::FfiConverter<crate::UniFfiTag>>::try_lift(r#ptr) {
+            Ok(ref val) => val,
+            Err(err) => panic!("Failed to convert arg '{}': {}", "ptr", err),
+        }.hash(&mut s);
```

Comment:
I think this should be fully-qualified (something like `<val as Hash>::hash(&val, &mut s)`) just to be sure there's no name conflict.  The same goes for magic methods where the signature matches.